geo/benches/sweep_line_intersection.rs

```diff
@@ -90,8 +89,9 @@ fn bench_dense_line_intersections(c: &mut Criterion) {
         // Sweep line algorithm
         group.bench_function("sweep", |b| {
             b.iter(|| {
-                let intersections: Vec<_> =
-                    NewSweepIntersections::<_>::from_iter(lines.iter().cloned()).collect();
+                let intersections: Vec<_> = Intersections::from_iter(lines.iter().cloned())
```

Comment:
Oh also very notable: I broke the API. 

Intersection no longer impl's iter, but you can **get** an iter.

The old API built up all the intersections into a Vec, and then its own iter implementation was based on that. The new API is lazy - it only computes the next intersection when asked. 

So if for some rason you want to do: `Intersections::new(segments).iter().take(3)` you don't waste any work.